src/twitch.rs

```diff
@@ -27,44 +24,48 @@ pub async fn twitch_irc(config: &CompleteConfig, tx: Sender<Data>, mut rx: Recei
     let mut stream = client.stream().unwrap();
 
     loop {
-        if let Some(Some(Ok(message))) = unconstrained(stream.next()).now_or_never() {
-            match message.command {
-                Command::PRIVMSG(ref _target, ref msg) => {
-                    let user = match message.source_nickname() {
-                        Some(username) => username.to_string(),
-                        None => "Undefined username".to_string(),
-                    };
-                    tx.send(Data::new(
-                        Local::now()
-                            .format(config.frontend.date_format.as_str())
-                            .to_string(),
-                        user,
-                        msg.to_string(),
-                        false,
-                    ))
-                    .await
-                    .unwrap();
-                }
-                Command::NOTICE(ref _target, ref msg) => {
-                    tx.send(Data::new(
-                        Local::now()
-                            .format(config.frontend.date_format.as_str())
-                            .to_string(),
-                        "twitch".to_string(),
-                        format!("NOTICE: {}", msg),
-                        true,
-                    ))
-                    .await
-                    .unwrap();
-                }
-                _ => (),
-            }
-        }
+        tokio::select! {
+            biased;
 
-        if let Some(Some(message)) = unconstrained(rx.recv()).now_or_never() {
-            client
+            Some(message) = rx.recv() => {
```

Comment:
Making sure we get a message through `recv` instead of `try_recv`, yeah?

Reply:
`select!` waits on multiple branches concurrently, so yes this won't block the twitch handler.

Reply:
Very cool.